js/webform_civicrm_payment.js

```diff
@@ -120,4 +127,18 @@ cj(function($) {
     });
 
   tally();
+
+  var payment = {
+    getTotalAmount: function() {
+      return getTotalAmount();
+    }
+  };
+
+  if (typeof CRM.payment === 'undefined') {
+    CRM.payment = payment;
+  }
+  else {
+    $.extend(CRM.payment, payment);
```

Comment:
At this stage CRM.payment seems to be a feature of mjwshared but I think we should discuss this being a core js class (how would that look?)- this approach would allow for backward compatibility - ie does the core version have the class

But, how would migrating it from mjwshared to core look? Would there be a class if both were present?

Reply:
Currently `CRM.payment` in mjwshared has a bunch of functions - I'm not sure if all of them are "core-ready" or not but `getTotalAmount` is the most obvious / simple of them.  The `CRM.payment = .. else $.extend(CRM.payment` construct is nice from a transition / improvement point of view because it allows anyone to be the definer of CRM.payment, whether that be webform, CiviCRM core, mjwshared etc. The only issue becomes load order - ie. which one actually gets used - this line (https://lab.civicrm.org/extensions/mjwshared/-/blob/0.9/js/crm.payment.js#L160) handles that in mjwshared and defers to the existing `CRM.payment.getTotalAmount` function. I would expect a core implementation to do the same so that the "final client", ie webform or anything that loads CiviCRM payment processors (eg. if we had an integration with caldera forms in the future) would be able to define their own implementation of `CRM.payment.getTotalAmount()` - the spec simply being "Return a numeric formatted total amount that should be charged".